CesiumGS/cesium
# 3d tiles height reference

To be merged after https://github.com/CesiumGS/cesium/pull/11581

Fixes https://github.com/CesiumGS/cesium/issues/7044
Fixes #9680
CC https://github.com/CesiumGS/cesium/issues/11544

Building on this previous PR, this allows more user-friendly APIs for clamping to 3D Tiles as well as or instead of terrain by extending the `HeightReference` types:
 - `CLAMP_TO_GROUND`/`RELATIVE_TO_GROUND` now clamps to or is relative to terrain or 3D Tiles, whichever is higher
 - Added `CLAMP_TO_TERRAIN`/`RELATIVE_TO_TERRAIN` to clamp to or position relative to terrain only
 - Added `CLAMP_TO_3D_TILE`/`RELATIVE_TO_3D_TILE` to clamp to or position relative to terrain only
 
 This was to align to the possible `ClassificationType` values, `TERRAIN`, `CESIUM_3D_TILE`, or `BOTH`.

I've replaced the sandcastle for the (now outdated) `scene.clampToHeight` method of clamping, which is now [Clamp Model to Ground](http://cesium-dev.s3-website-us-east-1.amazonaws.com/cesium/3d-tiles-height-reference/Apps/Sandcastle/index.html?src=Clamp%20Model%20to%20Ground.html). I've also extended this and the [Clamp Entities to Ground](http://cesium-dev.s3-website-us-east-1.amazonaws.com/cesium/3d-tiles-height-reference/Apps/Sandcastle/index.html?src=Clamp%20Entities%20to%20Ground.html) sandcastle a bit to showcase available options.

Testing Plan:
- Make sure all options are working on the new sandcastles, "Clamp Model to Ground" and "Clamp Entities to Ground"
- Confirm [data sources with `clampToGround` work on 3D Tiles](http://localhost:8080/Apps/Sandcastle/index.html#c=5VZtb9owEP4rJ7QPQaqMpu7DxGi1FVBVDcpU6KpqmVbjHGDh2JHj8FLEf98lgfASqn1cqyJIuPOd77nnzskJo2MHM4lztHABGufQxFgmIfuZ6Ty/IjK5abTjUqP1K2ew8jXAWJkh1mHEVYy+Xle/+NrXCh3MjVXBQCqM0ZHS2WXusK+nWHzOpdtGExa5w2tjxgp/TIwzJCsZOynOWxsXL40AG6wsFqiRRVaG0skZxowHgbcfIbVeg+BOTMBDa42t5jAEpWwUMmXG3lM7XQBleCD1GA4jw3kLss3A5Vuy1B3gwyrbbv2UhUizFhmLkdRXiVTBMZM/Cr2Xk9TnOhA8dgSCUA+MUUNuu6gTLw3wK4+yym8ADheuDn5lYLmYwly6CTzwZWSkdjEVY2tGEFChINNRooWTRoNX3e1SEBdwx/smsVTW3RpAxt++Arbwr6NFq/BhKVVHdkDgGKvRt8/DSGFqXRtHi5riodSc0d89nNvP6lhBlSGHaGCurUl0UAdnEyy5rQ8V1X3xQGBugtrbcbHLu3oce8PMszHhwOwZMtSOugvjvPMKCDtxvcG3vZeKdmcSh6+7SjaF+K6L9N+P0+nCzCO3KUu5Dv+uwvq1s95NlJOUL2QPtvhNPdmEScXFwE5fOjinKvTWzkV/GQ4NvSaX0ItSHO/wdUNkpN3ZJBpsfQsxk9hju9PpPZzwmG96+CbkYxqQdmMBG1kTdvlU3tDAcFPOK89tKMVSKDyRwx5LOYSrzn37pNWnz2Vt9U09pn9nV7/i8unIr/i6NDzZzSRZ6sBy75FtaGb4TanDMVIoI6b59Y7rMdJuW4YLHbu/verd37barRc8+w6jI8dUxfqP/UG7+6fZ6TW/p6Mi/SpnlUbslgovt2l/lWFkrIPEKo962iH1NI3CcW2YiCnNnCIuKGvU9l0bgZyBDC5OTOhpa8cxrYwSpfrymZrpslEj+5LrZvLtzdAqvkzNJh8vO7mSMdaokXjasyhMsZ5b/AU)
- Confirm nothing crashes when switching between 2D/3D/CV 

TODO:
- [x] Update CHANGES.md
- [x] Confirm 2D/CV mode
- [x] Sandcastle thumbnails
